chart/templates/etcd-statefulset.yaml

```diff
@@ -0,0 +1,174 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: etcd-{{ .Values.role }}-client
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: etcd-statefulset
+    role: {{ .Values.role }}
+spec:
+  type: ClusterIP
+  sessionAffinity: None
+  selector:
+    app: etcd-statefulset
+    role: {{ .Values.role }}
+  ports:
+  - name: client
+    protocol: TCP
+    port: 2379
+    targetPort: 2379
+---
+apiVersion: {{ include "statefulsetversion" . }}
+kind: StatefulSet
+metadata:
+  name: etcd-{{ .Values.role }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: etcd-statefulset
+    role: {{ .Values.role }}
+spec:
+  updateStrategy:
+    type: RollingUpdate
+  serviceName: etcd-{{.Values.role}}
+  replicas: {{ .Values.replicas }}
+  selector:
+    matchLabels:
+      app: etcd-statefulset
+      role: {{ .Values.role }}
+  template:
+    metadata:
+      annotations:
+        checksum/configmap-etcd-bootstrap-config: {{ include (print $.Template.BasePath "/configmap-etcd-bootstrap.yaml") . | sha256sum }}
+{{- if .Values.podAnnotations }}
```

Comment:
Currently i don't  find good reason to have this `if statement` be part of chart. Will have to look some standard chart example regarding flexibility of pod annotations.

Reply:
I'd prefer to keep this one.

A simple example that we'd need these annotations could be having prometheus a installed in the cluster (not the operator but the regular prometheus) and let it scrape etcd metrics using the some kubernetes service discovery which are usually configured via pod annotations.

Reply:
Ahh right.